demos/ota/ota_demo_core_http/ota_demo_core_http.c

```diff
@@ -0,0 +1,1620 @@
+ * statistics like number of packets received, dropped, processed and queued per connection.
+ */
+#define OTA_EXAMPLE_TASK_DELAY_MS                ( 1000U )
+
+/**
+ * @brief The maximum size of the file paths used in the demo.
+ */
+#define OTA_MAX_FILE_PATH_SIZE                   ( 260 )
+
+/**
+ * @brief The maximum size of the stream name required for downloading update file
+ * from streaming service.
+ */
+#define OTA_MAX_STREAM_NAME_SIZE                 ( 128 )
+
+/**
+ * @brief The maximum back-off delay (in milliseconds) for retrying connection to server.
+ */
+#define CONNECTION_RETRY_MAX_BACKOFF_DELAY_MS    ( 5000U )
+
+/**
+ * @brief The base back-off delay (in milliseconds) to use for connection retry attempts.
+ */
+#define CONNECTION_RETRY_BACKOFF_BASE_MS         ( 500U )
+
+/**
+ * @brief Number of milliseconds in a second.
+ */
+#define NUM_MILLISECONDS_IN_SECOND               ( 1000U )
+
+/**
+ * @brief Maximum size of the url.
+ */
+#define OTA_MAX_URL_SIZE                         ( 2048U )
+
+/**
+ * @brief Maximum size of the auth scheme.
+ */
+#define OTA_MAX_AUTH_SCHEME_SIZE                 ( 2048U )
+
+/**
+ * @brief Size of the network buffer to receive the MQTT message.
+ *
+ * The largest message size is data size from the AWS IoT streaming service,
+ * otaconfigFILE_BLOCK_SIZE + extra for headers.
+ */
+
+#define OTA_NETWORK_BUFFER_SIZE          ( otaconfigFILE_BLOCK_SIZE + OTA_MAX_URL_SIZE + 128 )
+
+/**
+ * @brief The maximum number of retries for connecting to server.
+ */
+#define CONNECTION_RETRY_MAX_ATTEMPTS    ( 5U )
+
+/* Check that size of the user buffer is defined. */
+#ifndef USER_BUFFER_LENGTH
+    #define USER_BUFFER_LENGTH    ( 2048 )
+#endif
+
+/**
+ * @brief The MQTT metrics string expected by AWS IoT.
+ */
+#define METRICS_STRING           "?SDK=" OS_NAME "&Version=" OS_VERSION "&Platform=" HARDWARE_PLATFORM_NAME "&OTALib=" OTA_LIB
+
+/**
+ * @brief The length of the MQTT metrics string expected by AWS IoT.
+ */
+#define METRICS_STRING_LENGTH    ( ( uint16_t ) ( sizeof( METRICS_STRING ) - 1 ) )
+
+
+#ifdef CLIENT_USERNAME
+
+/**
+ * @brief Append the username with the metrics string if #CLIENT_USERNAME is defined.
+ *
+ * This is to support both metrics reporting and username/password based client
+ * authentication by AWS IoT.
+ */
+    #define CLIENT_USERNAME_WITH_METRICS    CLIENT_USERNAME METRICS_STRING
+#endif
+
+/*-----------------------------------------------------------*/
+
+/* Linkage for error reporting. */
+extern int errno;
+
+/**
+ * @brief Struct for firmware version.
+ */
+const AppVersion32_t appFirmwareVersion =
+{
+    .u.x.major = APP_VERSION_MAJOR,
+    .u.x.minor = APP_VERSION_MINOR,
+    .u.x.build = APP_VERSION_BUILD,
+};
+
+/* Each compilation unit must define the NetworkContext struct. */
+struct NetworkContext
+{
+    OpensslParams_t * pParams;
+};
+
+/**
+ * @brief Network connection context used in this demo.
+ */
+static NetworkContext_t networkContext;
```

Comment:
```suggestion
static NetworkContext_t networkContextMqtt;
```

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1425/commits/9f18e9905580d6e4067eaa4a215da9dbe1e2453a